userguide/tutorials/plugin_development.adoc

```diff
@@ -189,6 +189,57 @@ When you start developing your own plugin, it would be useful to be able to set
 
 . This runs the application in debug mode. You can also set additional breakpoints as required.
 
+=== Authentication within plugins
+
+In order to invoke read/write API operations like `AccountUserApi#getAccountById` , plugin code must authenticate against Kill Bill first. Otherwise, it will result in an `org.apache.shiro.authz.UnauthenticatedException`. This section explains how authentication can be done.
+
+==== Authentication steps
+
+In order to authenticate against Kill Bill, the following needs to be done:
+
+1. Obtain credentials - Typically, plugins cannot use the Kill Bill credentials and should be configured with a custom username/password. You can see an example of retrieving custom credentials in the https://github.com/killbill/killbill-email-notifications-plugin/blob/8df47156a2e80c65ce574e0ad689afd02b926f59/src/main/java/org/killbill/billing/plugin/notification/setup/EmailNotificationListener.java#L230[EmailnotificationListener] code.
```

Comment:
`plugins cannot use the Kill Bill credentials` is a tad confusing. Any credentials should work, it's just that it's not best practice to use the `admin` credentials, to limit the scope of operations a plugin can do.

Reply:
Fixed